src/decoder/lattice-faster-decoder.h

```diff
@@ -413,21 +364,52 @@ class LatticeFasterDecoderTpl {
   // forward-cost[t] if there were no final-probs active on the final frame.
   // You cannot call this after FinalizeDecoding() has been called; in that
   // case you should get the answer from class-member variables.
-  void ComputeFinalCosts(unordered_map<Token*, BaseFloat> *final_costs,
+  void ComputeFinalCosts(unordered_map<const Token *, BaseFloat> *final_costs,
                          BaseFloat *final_relative_cost,
                          BaseFloat *final_best_cost) const;
 
+  // Lists of tokens, indexed by frame (members of TokenList are toks,
+  // must_prune_forward_links, must_prune_tokens).
+  std::vector<TokenList> active_toks_;
+
+  // This contains, for each frame, an offset that was added to the acoustic
+  // log-likelihoods on that frame in order to keep everything in a nice dynamic
+  // range i.e.  close to zero, to reduce roundoff errors.
+  std::vector<BaseFloat> cost_offsets_;
+
+  // decoding_finalized_ is true if someone called FinalizeDecoding().  [note,
+  // calling this is optional].  If true, it's forbidden to decode more.  Also,
+  // if this is set, then the output of ComputeFinalCosts() is in the next
+  // three variables.  The reason we need to do this is that after
+  // FinalizeDecoding() calls PruneTokensForFrame() for the final frame, some
+  // of the tokens on the last frame are freed, so we free the list from toks_
+  // to avoid having dangling pointers hanging around.
+  bool decoding_finalized_;
+
+  unordered_map<const Token *, BaseFloat> final_costs_;
+
+  KALDI_DISALLOW_COPY_AND_ASSIGN(LatticeFasterDecoderTpl);
```

Comment:
I'm not entirely positive but it looks like you moved this from a private: section of the class to a protected: section. Why?

Reply:
It has always been in protected not private. In fact, all methods were in protected before.